subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -143,6 +150,41 @@ int fp_keys_decrypt(const struct bt_conn *conn, uint8_t *out, const uint8_t *in)
 	return err;
 }
 
+int fp_keys_additional_data_encode(const struct bt_conn *conn, uint8_t *out, const uint8_t *data,
+				   size_t data_len, const uint8_t *nonce)
+{
+	struct fp_procedure *proc = &fp_procedures[bt_conn_index(conn)];
+	int err = 0;
+
+	if (!is_key_generated(proc->state)) {
+		err = -EACCES;
+	}
+
+	if (!err) {
+		err = fp_crypto_additional_data_encode(out, data, data_len, proc->aes_key, nonce);
+	}
+
+	return err;
+}
+
+int fp_keys_additional_data_decode(const struct bt_conn *conn, uint8_t *out_data,
+				   const uint8_t *in_packet, size_t packet_len)
+{
+	struct fp_procedure *proc = &fp_procedures[bt_conn_index(conn)];
+	int err = 0;
+
+	if (!is_key_generated(proc->state)) {
+		err = -EACCES;
+	}
+
+	if (!err) {
+		err = fp_crypto_additional_data_decode(out_data, in_packet, packet_len,
+						       proc->aes_key);
+	}
+
+	return err;
```

Comment:
Those functions are consistent with functions `fp_keys_encrypt` and `fp_keys_decrypt`. The assumption have been made that keys module is responsible for using keys and handle every operation related to it. We want to avoid passing keys between modules (moving those functions to `fp_gatt_service.c` would require adding also `fp_keys_get_key` API.